File: thrifts-dotnet/ThriftS.Serializer/ValueContainer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThriftS.Common;
using ThriftS.Common.Attributes;

namespace ThriftS.Serializer
{
    [ThriftSModel]
    internal class ValueContainer<T>
    {
        [ThriftSMember(1)]
        public T Value { get; set; }

        public ValueContainer()
        {
        }

        public ValueContainer(T value)
        {
            Value = value;
        }
    }
}

File: thrifts-java/thrifts-serializer/src/main/java/thrifts/serializer/ValueContainer.java
package thrifts.serializer;

import thrifts.common.annotations.ThriftSMember;
import thrifts.common.annotations.ThriftSModel;

@ThriftSModel
final class ValueContainer<T> {
    @ThriftSMember(tag = 1)
    private T value;

//    private Class<?> valueType;

    public ValueContainer() {
    }

    public ValueContainer(T value) {
        this.value = value;
//        this.valueType = value.getClass();
    }

    public T getValue() {
        return this.value;
    }

    public void setValue(T value) {
        this.value = value;
    }
}
